tx.go

```diff
@@ -123,6 +127,52 @@ func FromContext(ctx context.Context) (Tx, bool) {
 	return t, ok
 }
 
+type DoneFunc func() error
+
+// TraceReq traces an http request. It returns a new context with the transaction
+// included in it, and a func to be called when the request is finished.
+//
+// Usage:
+//
+//     ctx, done := TraceRequest(ctx, name, req)
+//     defer done()
+func TraceRequest(ctx context.Context, name string, req *http.Request) (context.Context, DoneFunc, error) {
+	tx := NewRequestTx(name, req.URL.String())
+	ctx = WithTx(ctx, tx)
+	err := tx.Start()
+	df := func() error {
+		return tx.End()
+	}
+
+	return ctx, df, err
```

Comment:
Actually, it'd probably be better to return an interface that has `Done()` and `Err()` methods.

That way:

``` go
t := TraceRequest(...)
defer function() {
    t.Done()
    if t.Err() != nil {
        // log error
    }
}()
```